src/codemodder/semgrep.py

```diff
@@ -71,7 +71,29 @@ def from_sarif(
         for location in sarif_result["locations"]:
             artifact_location = SemgrepLocation.from_sarif(location)
             locations.append(artifact_location)
-        return cls(rule_id=rule_id, locations=locations)
+        all_flows: list[list[Location]] = [
```

Comment:
Is the only difference between this and CodeQL the location class? Seems like a good opportunity for some kind of generics pattern but it's not that important.

Reply:
For now yes, at least this particular part (codeflows) are the same. They both follow the spec:
https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/sarif-v2.1.0-errata01-os-complete.html#_Toc141790990

I think we may be able to generalize all of them with a single class parameterized by location objects. I'll write a ticket for it.